### PR TITLE
Add release process documentation

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:
This adds a new release process documentation, that other people are
able to cut releases for this repository, too.
#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: release.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+A new seccomp-operator release can be done by overall three Pull Requests (PRs).
+Please ensure that no other PRs got merged in between. The overall process
+should not take longer than a couple of minutes, but it is required to have one
+of the repository [owners](./OWNERS) at hand to be able to merge the PRs.
+
+The first PR targets this repository and:
+
+- bumps the [`VERSION`](VERSION) file to the target version
+- changes the `image` names in all files of [./deploy](deploy) from
```
Comment (hasheddan) [inline release.md:11]:
```suggestion
- and changes the `image` names in all files of [./deploy](deploy) from
```

File: release.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+If this PR got merged, then we're finally ready to [create the
+release](https://github.com/kubernetes-sigs/seccomp-operator/releases/new)
+directly on GitHub and add the release notes. The release notes can be generated
+by the [official Kubernetes Release Notes
+tool](https://github.com/kubernetes/release/tree/master/cmd/release-notes).
+
+After that, another PR against this repository has to be created, which:
+
+- bumps the [`VERSION`](VERSION) file to the current one, but now including the
```
Comment (hasheddan) [inline release.md:48]:
```suggestion
- bumps the [`VERSION`](VERSION) file to the next minor version, but now including the
```

Comment (hasheddan) [reply release.md:48]: If we did a patch release, would we change the VERSION to the patch (i.e. v0.1.1) then back to v0.2.0-dev after release?

File: release.md (hunk 1 above)
Comment (saschagrunert) [reply release.md]: Fixed :thu

File: release.md (hunk 2 above)
Comment (saschagrunert) [reply release.md]: Yes, this makes more sense. Fixed that. :+1:

**Pull request merged**